src/main/kotlin/org/openrewrite/kotlin/KotlinTypeMapping.kt

```diff
@@ -957,8 +982,7 @@ class KotlinTypeMapping(
         clazz = if (type.classifier != null) {
             TypeUtils.asFullyQualified(type(type.classifier!!))
         } else {
-            val fq : JavaType = buildType(type.classifierQualifiedName)
-            fq as FullyQualified
+            createShallowClass(type.classifierQualifiedName)
```

Comment:
Info:
`buildType(type.classifierQualifiedName)` either creates a `JavaType.Primitive` or a `ShallowClass`, and a primitive is never valid. So, we use `createShallowClass` to create a `ShallowClass`, which also caches the signature for the type.